app/controllers/green_lanes/results_controller.rb

```diff
@@ -2,66 +2,67 @@ module GreenLanes
   class ResultsController < ApplicationController
     include GreenLanesHelper
 
-    before_action :check_green_lanes_enabled,
-                  :disable_switch_service_banner,
-                  :disable_search_form
+    before_action :set_up, only: :show
 
     def show
-      goods_nomenclature = GreenLanes::GoodsNomenclature.find(
-        results_params[:commodity_code],
-        { filter: {
-          geographical_area_id: results_params[:country_of_origin],
-          moving_date: results_params[:moving_date],
-        } },
-        { authorization: TradeTariffFrontend.green_lanes_api_token },
-      )
+      @next_page = determine_next_page
+    end
+
+    private
+
+    def set_up
+      check_green_lanes_enabled
+      disable_switch_service_banner
+      disable_search_form
+      goods_nomenclature
+      set_instance_variables
+      determine_assessments
+    end
+
+    def results_params
+      params.permit(:commodity_code, :country_of_origin, :moving_date, :c1ex, :c2ex, :category, :ans)
+    end
+
+    def goods_nomenclature
+      @goods_nomenclature ||= GreenLanes::FetchGoodsNomenclature.new(results_params).call
+    end
 
-      @commodity_code = goods_nomenclature.goods_nomenclature_item_id
+    def set_instance_variables
+      @commodity_code = @goods_nomenclature.goods_nomenclature_item_id
       @country_of_origin = results_params[:country_of_origin] || GeographicalArea::ERGA_OMNES
       @country_description = GeographicalArea.find(@country_of_origin).description
       @moving_date = results_params[:moving_date]
-
-      next_page = DetermineNextPage
-        .new(goods_nomenclature)
-        .next(
-          cat_1_exemptions_apply:,
-          cat_2_exemptions_apply:,
-        )
-
-      case next_page
-      when :result_cat_1
-        render 'result_cat_1'
-      when :result_cat_2
-        render 'result_cat_2'
-      when :result_cat_3
-        render 'result_cat_3'
-      end
+      @category = results_params[:category]
+      @ans = JSON.parse(results_params[:ans])
     end
 
-    private
+    def determine_assessments
+      assessments = GreenLanes::DetermineAssessments.new(@goods_nomenclature, @ans)
```

Comment:
We've got a lot of dependency coupling to the presenter, here, in the controller. I'm wondering whether its cleaner to just memoize the presenter and have the view ask questions of it?